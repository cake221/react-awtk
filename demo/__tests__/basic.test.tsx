import "../basic"

describe("test basic.tsx", ()=>{

  it("Snapshot Test", ()=>{
    expect(__AwtkSnapshot).toMatchSnapshot();
  })
})
