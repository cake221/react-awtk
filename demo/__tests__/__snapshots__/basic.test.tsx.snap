// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test basic.tsx Snapshot Test 1`] = `
Array [
  "getRootHostContext",
  "getChildHostContext",
  "shouldSetTextContent",
  "createInstance",
  "funtion  window_open( name ) :basic",
  "funtion  window_set_fullscreen( widget, fullscreen ) :undefined, undefined",
  "finalizeInitialChildren",
  "prepareForCommit",
  "appendNodeChild",
  "resetAfterCommit",
  "getPublicInstance",
  "funtion  widget_lookup( widget, name, recursive ) :undefined, inc_value, true",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_CLICK, function () {
            _this.winRef.current.child(\\"bar1\\").addValue(10);
            _this.winRef.current.child(\\"bar2\\").addValue(10);
        }, null",
  "funtion  widget_lookup( widget, name, recursive ) :undefined, dec_value, true",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_CLICK, function () {
            _this.winRef.current.child(\\"bar1\\").addValue(-10);
            _this.winRef.current.child(\\"bar2\\").addValue(-10);
        }, null",
  "funtion  widget_lookup( widget, name, recursive ) :undefined, close, true",
  "funtion  widget_on( widget, type, on_event, ctx ) :undefined, EVT_CLICK, function () {
            var bar1 = _this.winRef.current.child(\\"bar1\\").getValue();
            var bar2 = _this.winRef.current.child(\\"bar2\\").getValue();
            console.log('bar1 value:' + bar1);
            console.log('bar2 value:' + bar2);
        }, null",
  "funtion  widget_get_window_manager( widget ) :undefined",
  "funtion  window_manager_cast( widget ) :[object Object]",
  "funtion  window_manager_set_show_fps( widget, show_fps ) :undefined, true",
]
`;
